banded_color_bar.coffee

```diff
@@ -0,0 +1,65 @@
+ColorBar = require "models/annotations/color_bar"
+LinearColorMapper = require "models/mappers/linear_color_mapper"
+p = require "core/properties"
+
+class BandedColorBarView extends ColorBar.View
+  bind_bokeh_events: () ->
+    @listenTo(@model, 'change:band_start', () ->
+      @_set_canvas_image()
+      @plot_view.request_render())
+    @listenTo(@model, 'change:band_end', () ->
+      @_set_canvas_image()
+      @plot_view.request_render())
+
+  _set_canvas_image: () ->
+    palette = @model.color_mapper.palette
+
+    if @model.orientation == 'vertical'
+      palette = palette.slice(0).reverse()
+
+    switch @model.orientation
+      when "vertical" then [w, h] = [1, palette.length]
+      when "horizontal" then [w, h] = [palette.length, 1]
```

Comment:
`switch` is an expression in coffeescript, so you can write `[w, h] = switch ...` instead. Not only this tells the intent better, it also would allow to use `const` keyword if such existed in coffeescript (or if we every switch to typescript, but then no `switch` expressions).